### PR TITLE
PasswordDialog: add CAPSLOCK checking

### LINKED ISSUE #881: [UX] Caps lock warning on wallet password window
Show a warning that caps lock is enabled on wallet password window.

### PR DESCRIPTION
**Pull request by mmbyday**
Fixes #881 
Improving user experience / usability.

paass

### CODE REVIEW

**Comment by selsta** [comment]
I like the idea but I wonder if there’s an easier way to add this.

File: js/Utils.js [hunk 1]
```diff
@@ -48,4 +48,42 @@ function showSeedPage() {
...
+function isalpha(letter){
+    var alpha = String("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+    var i;
+    for(i in alpha )
+        if(letter == alpha[i])
+            return true;
+    return false;
+}
+
+function isLowerCaseChar(letter){
```
Comment (xmrdsc) [inline js/Utils.js]: `function isLowerCaseChar(letter){ return letter === letter.toLowerCase(); }`

**Comment by xmrdsc** [comment]
Forget my review, I suggest a different approach.

Add a keyfilter here:

https://github.com/monero-project/monero-gui/blob/773c8370d8cdc2d6a17c7c493dadd99f924a42c2/filter.cpp#L40-L43

Emit a new signal here:

https://github.com/monero-project/monero-gui/blob/773c8370d8cdc2d6a17c7c493dadd99f924a42c2/main.cpp#L325

Call the filter 'capslockToggle'.

Listen for it in `main.qml` and set an appWindow 'global' variable:

```js
function capslockToggle(state) {
    appWindow.capslockToggled = state;
}
```

Then in password dialog you can use `appWindow.capslockToggled`.

Superior solution because it would indicate capslock usage without having to first enter a character.

**Comment by mmbyday** [comment]
Good idea, will wait for #1814 to be merged first before implementing this superior event/signal based approach ;-)

**Comment by xmrdsc** [comment]
Do you want to give it a shot? I can pick this up too if you want :)

**Comment by mmbyday** [comment]
@xmrdc I tried, but will need your help. Continue here, https://github.com/mmbyday/monero-gui/pull/2 , or start fresh :)

**Comment by xmrdsc** [comment]
I looked into this. I got the signals to work, however, detecting capslock status is quite a pain to make in a cross-platform manner. For example, on OSX a capslock press is exactly that, a press - it and stays pressed until you click it again, at which point capslock releases - you'll end up with only 1 keydown event. Thus I am in favor of rebasing this ;)

File: js/Utils.js (hunk 1 above)
Comment (mmbyday) [reply js/Utils.js]: Awesome 👍

File: js/Utils.js [hunk 2]
```diff
@@ -48,4 +48,42 @@ function showSeedPage() {
...
     updateBalance();
-}
\ No newline at end of file
+}
+
+/*******************
+   CheckCapsLocks
+   author qyvlik email qyvlik@qq.com https://github.com/qyvlik/CheckCapsLock.qml/blob/master/CheckCapsLock.qml
+*******************/
+function isalpha(letter){
```
[the remark this answers is not in the document]
Comment (mmbyday) [reply js/Utils.js]: Awesome 👍 👍

**Comment by mmbyday** [comment]
Rebased.

**Pull request merged**